components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/service/IdentityKeyStoreGeneratorImpl.java

```diff
@@ -0,0 +1,334 @@
+            int keySize = getKeySize(keyGenerationAlgorithm);
+            if (keySize != 0) {
+                keyPairGenerator.initialize(keySize);
+            }
+            KeyPair keyPair = keyPairGenerator.generateKeyPair();
+
+            // Common Name and alias for the generated certificate
+            String commonName = "CN=" + tenantDomain +
+                    KEY_STORE_CONTEXT_SEPARATOR + context + ", OU=None, O=None, L=None, C=None";
+
+            //generate certificates
+            X500Name distinguishedName = new X500Name(commonName);
+
+            Date notBefore = new Date(System.currentTimeMillis() - 1000L * 60 * 60 * 24 * 30);
+            Date notAfter = new Date(System.currentTimeMillis() + (1000L * 60 * 60 * 24 * 365 * 10));
+
+            SubjectPublicKeyInfo subPubKeyInfo = SubjectPublicKeyInfo.getInstance(keyPair.getPublic().getEncoded());
+            BigInteger serialNumber = BigInteger.valueOf(new SecureRandom().nextInt());
+
+            X509v3CertificateBuilder certificateBuilder = new X509v3CertificateBuilder(
+                    distinguishedName,
+                    serialNumber,
+                    notBefore,
+                    notAfter,
+                    distinguishedName,
+                    subPubKeyInfo
+            );
+
+            String algorithmName = getSignatureAlgorithm();
+            JcaContentSignerBuilder signerBuilder =
+                    new JcaContentSignerBuilder(algorithmName).setProvider(getJCEProvider());
+            PrivateKey privateKey = keyPair.getPrivate();
+            X509Certificate x509Cert = new JcaX509CertificateConverter().setProvider(getJCEProvider())
+                    .getCertificate(certificateBuilder.build(signerBuilder.build(privateKey)));
+
+            //add private key to KS
+            keyStore.setKeyEntry(tenantDomain +KEY_STORE_CONTEXT_SEPARATOR + context,
+                    keyPair.getPrivate(), password.toCharArray(),
+                    new java.security.cert.Certificate[]{x509Cert});
+        } catch (Exception ex) {
+            String msg = "Error while generating the Context certificate for tenant :" +
+                    tenantDomain + ".";
+            LOG.error(msg, ex);
+            throw new KeyStoreManagementException(msg, ex);
+        }
+    }
+
+    private static String getKeyGenerationAlgorithm() {
+
+        String signatureAlgorithm = getSignatureAlgorithm();
+        // If the algorithm naming format is {digest}with{encryption}, we need to extract the encryption part.
+        int withIndex = signatureAlgorithm.indexOf("with");
+        if (withIndex != -1 && withIndex + 4 < signatureAlgorithm.length()) {
+            return signatureAlgorithm.substring(withIndex + 4);
+        } else {
+            // The algorithm name is same as the encryption algorithm.
+            // This need to be updated if more algorithms are supported.
+            return signatureAlgorithm;
```

Comment:
Might be able to use StringUtils split to directly get this done

Reply:
addressed in https://github.com/wso2/carbon-identity-framework/pull/6189/commits/6ca0b3d01b63c8f3f8ee122988af923b204f342b